### ISSUE TITLE
traffic disruption after smithproxy restart if policy "features" feature is used

### ISSUE THREAD

**Issue by astibal** [astibal/smithproxy#40]
# Overview
Traffic disruption after smithproxy restart if policy "features" feature is used.

Smithproxy policies may be configured with "feature" feature. 
These "features" apply specific conditions to policy traffic, which are not supposed to be parts of main policy configuration.

To date of writing this report, these features are:
- `sink-left` - all data from client to server are consumed and not forwarded\
- `sink-right` - consumed from server
- `sink-all` - consumed from both directions
- `statistics` - calculate flow statistical data, ie, entropy, various ratios

# Symptom timeline

1. When "features" are set in CLI, they are immediately working, no issue observed
2.  User saves the config with `save config` command, "features" are saved as an empty string, no traffic disruption, but configuration for the particular policy is already incorrect
3. User restarts proxy, or reload config by other means - **traffic on policy with "features" is disrupted,** policy is disabled
  This can be seen using `diag proxy policy list`, similar to this:
  ```
smithproxy(test)# diag proxy policy list 
 
PolicyRule: [udp] [Cidr: 0.0.0.0/0][Cidr: ::/0]:(*) -> [Cidr: 0.0.0.0/0][Cidr: ::/0]:(*) = ACCEPT(nonat): 
    auth=resolve
    tls=default
    det=detect
    cont=default
    alg_dns=dns_default

PolicyRule:  -- DISABLED -- [tcp] [Cidr: 0.0.0.0/0][Cidr: ::/0]:(*) -> [Cidr: 0.0.0.0/0][Cidr: ::/0]:(*) = ACCEPT(nonat): 
    auth=resolve
    tls=default
    det=detect
    cont=default
    alg_dns=dns_default 
  ```
As you see, second policy is **DISABLED**.

# Remedy

To immediately fix the issue, edit manually `/etc/smithproxy/smithproxy.cfg`, and fix lines:
```
    features = [ "" ]
```
to ie.
```
    features = [ "statistics" ]
```

... and **restart proxy service**. You can eventually issue `execute reload`, it may work, but was not tested.

# Prevention until fix is released
A. Recommended - after each "config save", please fix your config as indicated in Remedy section
or 
B. don't use features (not applicable in some scenarios)

**Comment by astibal** [comment]
should be fixed by c7ca3577

**Comment by astibal** [comment]
Issue is fixed. CLI has been improved to prevent some of config errors to be silently processed.
Now, when error occurs, CLI prompt contains `<!>` and configuration cannot be saved.
